Assets/MixedRealityToolkit.SDK/Features/UX/Scripts/PressableButtons/PressableButton.cs

```diff
@@ -264,18 +267,21 @@ private void Update()
 
         #region IMixedRealityTouchHandler implementation
 
-        private void PulseProximityLight(HandTrackingInputEventData eventData)
+        private void PulseProximityLight()
         {
             // Pulse each proximity light on pointer cursors' interacting with this button.
-            foreach (var pointer in eventData.InputSource.Pointers)
+            if (currentInputSources.Count != 0)
             {
-                ProximityLight[] proximityLights = pointer.BaseCursor?.GameObjectReference?.GetComponentsInChildren<ProximityLight>();
-
-                if (proximityLights != null)
+                foreach (var pointer in currentInputSources[currentInputSources.Count - 1].Pointers)
                 {
-                    foreach (var proximityLight in proximityLights)
+                    ProximityLight[] proximityLights = pointer.BaseCursor?.GameObjectReference?.GetComponentsInChildren<ProximityLight>();
```

Comment:
For a future PR: I wish that we could decouple movement / algorithms from visuals like light pulsing. Would have preferrred to have a LightPulse component that listens to when the button is pressed to play a pulse. That way you could customize / change when the pulse plays without needing to modify pressablebutton.cs.

Reply:
I wish we could too! But, sadly the ButtonPressed event doesn't pass any state along to know which input source fired the event.

Reply:
Good point. I created #5201 to track fixing this.